### DISCUSSION TITLE
Dev server fails on start, Error: "No matching export in "node_modules/remix/esm/index.js" for import "Link"

### DISCUSSION THREAD

**Discussion by protofarer** [remix-run/remix#2658] [Q&A] [answered]
I had a working dev build under remix 1.1 initially based off the devblog tutorial code and just upgraded to 1.3 which is when things broke. Steps:

1. Start new project using indie-stack template
2. Copy/paste directories over from previous version project: `./app`  `./prisma` etc...
3. Cull and modify files as needed
4. Build containers
5. `docker-compose up`

I can run the dev server without issue locally but when the server is run inside a container it fails to start, indeed erroring out on every single import statement under `./app`, eg:

stdout:
```
Building Remix app in production mode...
Built in 437ms
Removing intermediate container 2106c45708c7
 ---> 419c70e068a3
Step 11/13 : RUN chown -R 1000:1000 .
 ---> Running in 6d5ac36b2a57
Removing intermediate container 6d5ac36b2a57
 ---> 1b2ea07411a2
Step 12/13 : EXPOSE 3000
 ---> Running in 7e947c7194c9
Removing intermediate container 7e947c7194c9
 ---> e4fe431e8566
Step 13/13 : ENTRYPOINT ["/usr/bin/dumb-init", "--"]
 ---> Running in b889c0bb2670
Removing intermediate container b889c0bb2670
 ---> b18a404fc9f4
Successfully built b18a404fc9f4
Successfully tagged kenny.net-app-dev:latest
c-dbdev is up-to-date
Recreating c-appdev ... done
Attaching to c-dbdev, c-appdev
c-appdev | 
c-appdev | > dev
c-appdev | > NODE_ENV=development remix dev
c-appdev | 
c-appdev | Loading environment variables from .env
c-appdev | Watching Remix app in development mode...
c-appdev | 
c-appdev | ✘ [ERROR] No matching export in "node_modules/remix/esm/index.js" for import "redirect"
c-appdev | 
c-appdev |     app/admin.ts:2:9:
c-appdev |       2 │ import { redirect } from "remix";
c-appdev |         ╵          ~~~~~~~~
c-appdev | 
c-appdev |  ✘ [ERROR] No matching export in "node_modules/remix/esm/index.js" for import "Link"
c-appdev | 
c-appdev |     app/components/ClickyPost.tsx:1:9:
c-appdev |       1 │ import { Link } from 'remix';
c-appdev |         ╵          ~~~~
c-appdev | 
```

Appreciate any help!

**Comment by protofarer** [comment]
I've narrowed it down to `remix setup` stdout'ing a success during container build but thereafter shelling into the running app container shows `node_modules/remix/index*` without the expected exports as should have been done by `remix setup`. This may be a permissions issue related to container being setup as root and run as `USER node`. More to come.

Someone can let me know if this issue is out of context for remix discussion. Thanks.

**Comment by protofarer** [comment, marked as answer]
The dev server works now. However, I did not find the exact issue with the container, the problem seemed to have fixed itself after I built the container without specifying `USER node` then built container again with original Dockerfile and it worked. The fault seems to lie in how docker was caching the build layers and not starting the container with the `remix setup`'ed files.

**Comment by dakota002** [comment]
I had the same issue recently and was able to resolve it by deleting the .cache folder in my project
